### ISSUE TITLE
GPT OSS 120B on galaxy codegen fails with OOM

### ISSUE THREAD

**Issue by dgolubovicTT** [tenstorrent/tt-xla#4607]
`codegen_py` of `test_gpt_oss_120b_tp_galaxy_batch_size_64 --accuracy-testing` emits a complete 4-graph artifact (~872 GB on disk) , but the emitted code OOMs at runtime. When executed on the same Galaxy 4×8 regular test(non-codegen) path runs without OOM.

## Failure

```
TT_FATAL: Out of Memory: Not enough space to allocate 377487360 B DRAM buffer
across 12 banks, where each bank needs to store 31457280 B, but bank size is
1071821792 B (allocated: 1017920704 B, free: 53901088 B,
largest free block: 31453440 B) (assert.hpp:104)
TT_FATAL @ tt_metal/impl/allocator/bank_manager.cpp:462: false
```

## Repro steps

checkout: dgolubovic/autosearch-mp-gpt-oss-120b

The branch already carries:
- `f7005f9a5` — `test_gpt_oss_120b_tp_galaxy_batch_size_64` with the user-confirmed production weight overrides
- `8867a0489` + `9d0e086c5` — `CODEGEN_EXPORT_PATH` env-var hook in `benchmark_llm_torch_xla` (adds `backend=codegen_py, export_path, export_tensors=True, dry_run=True` when the env var is set; no-op otherwise)

Run: 
```
# Now trigger codegen of the same test. ~12 hours wall, ~872 GB on disk.
docker exec --user $(id -u):$(id -g) \
  --workdir /home/dgolubovic/repos/tt-xla \
  tt-xla-ird-$USER bash -lc '
    source venv/activate && cd tests/benchmark && \
    rm -rf /home/dgolubovic/repos/tt-xla/codegen_artifacts/gpt_oss_120b_full
    mkdir -p /home/dgolubovic/repos/tt-xla/codegen_artifacts
    export CODEGEN_EXPORT_PATH=/home/dgolubovic/repos/tt-xla/codegen_artifacts/gpt_oss_120b_full
    python -m pytest -svv \
      test_llms.py::test_gpt_oss_120b_tp_galaxy_batch_size_64 \
      --accuracy-testing
  '

# Codegen will finish exit 0. To reproduce the OOM, run the emitted
# graph_2 (logits prefill) via the harness on this branch:
docker exec --user $(id -u):$(id -g) \
  --workdir /home/dgolubovic/repos/tt-xla \
  tt-xla-ird-$USER bash -lc '
    source venv/activate && \
    python tests/benchmark/scripts/run_codegen_decode.py \
      /home/dgolubovic/repos/tt-xla/codegen_artifacts/gpt_oss_120b_full \
      --graphs graph_2
  '
```

The same test without codegen works end to end:
```
git checkout dgolubovic/autosearch-mp-gpt-oss-120b
# Verify the prod path still runs cleanly (no env var) — should produce
# TOP1=81.25% / TOP5=95.31% in ~37 min. Tested on this branch:
docker exec --user $(id -u):$(id -g) \
  --workdir /home/dgolubovic/repos/tt-xla \
  tt-xla-ird-$USER bash -lc '
    source venv/activate && cd tests/benchmark && \
    python -m pytest -svv \
      test_llms.py::test_gpt_oss_120b_tp_galaxy_batch_size_64 \
      --accuracy-testing
  '
```

**Comment by dgolubovicTT** [comment]
[repro_log.txt](https://github.com/user-attachments/files/27600421/repro_log.txt)

Here is the repro log for OOM when I run prefil graph (graph_0 or graph_2 they are essentially the same)
